### PR TITLE
docs: add generated API reference

### PR DESCRIPTION
**Pull request by jpeach**
Add some site scripts to generate API reference documentation for
the Contour CRDs. Unfortunately, we need to vendor the Go templates
from the documentation generator in order to add enough CSS to make
the results attractive.

Note, that if running the generator has the side-effect of updating
Go modules. These changes need to be reverted afterwards.

Signed-off-by: James Peach <jpeach@vmware.com>

### CODE REVIEW

File: site/docs/master/api.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Contour API Reference
+
```
Comment (jonasrosland) [inline site/docs/master/api.md]: Include something about how this is generated, which commands are used, etc.

Comment (jpeach) [reply site/docs/master/api.md]: Do you mean, how does a developer re-generate this?

**Comment by jpeach** [comment]
> Should these be part of the HTTPProxy docs/section? Right now they are disjointed from the others.

Maybe it makes sense under "Configuration"? Once we have API docs, I'd like to discuss tearing the large HTTPProxy document into smaller task-centric documents.

**Comment by stevesloka** [comment]
I 100% agree we should break apart the HTTPProxy docs, I'm thinking these API docs are just a piece of that.

File: site/docs/master/api.md (hunk 1 above)
Comment (jpeach) [reply site/docs/master/api.md]: There's a documented makefile target that generates these docs. If we find later that we need more info, I'll be happy to add it in a separate PR.

**Pull request merged**